server/tpm/js_ek_tpm_windows.go

```diff
@@ -0,0 +1,283 @@
+		return nil, nil, err
+	}
+
+	keysJSON, err := os.ReadFile(filename)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to read persisted keys from %q: %v", filename, err)
+	}
+
+	var tpmKeys natsTPMPersistedKeys
+	if err := json.Unmarshal(keysJSON, &tpmKeys); err != nil {
+		return nil, nil, fmt.Errorf("unable to unmarshal TPM file keys JSON from %s: %v", filename, err)
+	}
+
+	// Placeholder for future-proofing. Here is where we would
+	// check the current version against tpmKeys.Version and
+	// handle any changes.
+
+	// Base64 decode the private and public blobs.
+	publicBlob, err := base64.StdEncoding.DecodeString(string(tpmKeys.PublicKey))
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to decode publicBlob from base64: %v", err)
+	}
+	privateBlob, err := base64.StdEncoding.DecodeString(string(tpmKeys.PrivateKey))
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to decode privateBlob from base64: %v", err)
+	}
+	return publicBlob, privateBlob, nil
+}
+
+// Creates a new JetStream encryption key, seals it to the TPM, and saves the public and
+// private blobs to disk in a JSON encoded file. The key is returned as a string.
+func createAndSealJsEncryptionKey(rwc io.ReadWriteCloser, srkHandle tpmutil.Handle, srkPassword, jsKeyFile, jsKeyPassword string, pcr int) (string, error) {
+	// Get the authorization policy that will protect the data to be sealed
+	sessHandle, policy, err := policyPCRPasswordSession(rwc, pcr)
+	if err != nil {
+		return "", fmt.Errorf("unable to get policy: %v", err)
+	}
+	if err := tpm2.FlushContext(rwc, sessHandle); err != nil {
+		return "", fmt.Errorf("unable to flush session: %v", err)
+	}
+	// Seal the data to the parent key and the policy
+	user, err := nkeys.CreateUser()
+	if err != nil {
+		return "", fmt.Errorf("unable to create seed: %v", err)
+	}
+	// We'll use the seed to represent the encrpytion key.
+	jsStoreKey, err := user.Seed()
+	if err != nil {
+		return "", fmt.Errorf("unable to get seed: %v", err)
+	}
+	privateArea, publicArea, err := tpm2.Seal(rwc, srkHandle, srkPassword, jsKeyPassword, policy, jsStoreKey)
+	if err != nil {
+		return "", fmt.Errorf("unable to seal data: %v", err)
+	}
+	err = writeTPMKeysToFile(jsKeyFile, privateArea, publicArea)
+	if err != nil {
+		return "", fmt.Errorf("unable to write key file: %v", err)
+	}
+	return string(jsStoreKey), nil
+}
+
+// Unseals the JetStream encryption key from the TPM with the provided keys.
+// The key is returned as a string.
+func unsealJsEncrpytionKey(rwc io.ReadWriteCloser, pcr int, srkHandle tpmutil.Handle, srkPassword, objectPassword string, publicBlob, privateBlob []byte) (string, error) {
+	// Load the public/private blobs into the TPM for decryption.
+	objectHandle, _, err := tpm2.Load(rwc, srkHandle, srkPassword, publicBlob, privateBlob)
+	if err != nil {
+		return "", fmt.Errorf("unable to load data: %v", err)
+	}
+	defer func() {
+		tpm2.FlushContext(rwc, objectHandle)

```

Comment:
Another nit but these can be single-lined: `defer tpm2.FlushContext(rwc, objectHandle)`